# Python Tutorial

Dieses Python Tutorial behandelt nur die wesentlichen Informationen um Daten zu analysieren und zu visualisieren. Für weitergehende Informationen wird das offizielle (englischsprachige) [The Python Tutorial](https://docs.python.org/3/tutorial/) empfohlen.

## Python Code in Jupyter Notebooks

Jupyter Notebooks unterstützen die direkte Ausführung von Python Code. Daher können Beispiele wie das folgende direkt ausgeführt werden, indem die entsprechende Zelle gestartet (*run*) wird:

In [13]:
print("Hallo Welt!")

Hallo Welt!


## Kommentare

Es ist oft sehr hilfreich und es wird an vielen Stellen wärmstens empfohlen den Python Code so gut zu dokumentieren, dass andere ihn einfach verstehen können.

Kommentare beginnen mit dem # Zeichen und gehen bis zum ende einer Zeile. Kommentare können am Anfang einer Zeile oder nach einem Leerzeichen genutzt werden.

In [14]:
print("Hallo Welt!") # Hier beginnt der Kommentar
#print("Hallo Kommentar!") Dies ist ein Kommentar am Beginn der Zeile und der Code nach dem # wird nicht ausgeführt

Hallo Welt!


## Datentypen und Variablen

Python unterstützt verschiedene Datentypen, wobei wir nur folgende [integrierte Datentypen](https://docs.python.org/3/library/stdtypes.html) genauer beschreiben werden:

- Nummern
- Strings
- Listen

### Nummern

Python kann ohne weiter Kenntnisse als Taschenrechner eingesetzt werden:

In [15]:
1 + 2 * 3 / 4

2.5

Python unterscheidet hierbei zwischen folgenden Zahlentypen:
- `int` (Integer) - [Ganze Zahlen](https://de.wikipedia.org/wiki/Ganze_Zahl) z.B. `-5` oder `10`
- `float` - [Reelle Zahl](https://de.wikipedia.org/wiki/Reelle_Zahl) z.B. `-9.9` oder `2.5`
- `complex` - [Komplexe Zahl](https://de.wikipedia.org/wiki/Komplexe_Zahl) z.B. `complex('1+2j')` - diese spielen aber im folgenden keine Rolle und können ignoriert werden

### Strings

### Listen

### Variablen